model/group_allocator.go

```diff
@@ -0,0 +1,63 @@
+package model
+
+import (
+	"math/rand"
+)
+
+// InstallationGroupAllocator defines a way to allocate installations into groups based on
+// each allocator internal logic.
+type InstallationGroupAllocator interface {
+	Choose(groups []*GroupDTO) (*GroupDTO, error)
+}
+
+// RandomInstallationGroupAllocator allocates the installation in a random group
+type RandomInstallationGroupAllocator struct{}
+
+func (iga *RandomInstallationGroupAllocator) Choose(groups []*GroupDTO) (*GroupDTO, error) {
+	return groups[rand.Intn(len(groups))], nil
+}
+
+// NewRandomInstallationGroupAllocator
+func NewRandomInstallationGroupAllocator() InstallationGroupAllocator {
+	return &RandomInstallationGroupAllocator{}
+}
+
+// LowestCountInstallationGroupAllocator allocates the installation in the group that has the lowest
+// count for the defined state
+type LowestCountInstallationGroupAllocator struct {
+	// the state to check
+	installationState string
+}
+
+// getStateValue gets the value for the selected state to check from the provided group
+// defaults to the total instalations silently if the provided state is not handled
+func (iga *LowestCountInstallationGroupAllocator) getStateValue(group *GroupDTO) int64 {
+	switch iga.installationState {
+	case "updated":
+		return group.Status.InstallationsUpdated
+	default:
+		return group.Status.InstallationsTotal
+	}
+}
```

Comment:
After thinking this over a bit, let's keep things simple for now and just always use `Status.InstallationsTotal` for selecting the lowest count. I tried to think through if we may ever want to prioritize a certain state, but that seems pretty unlikely. If we ever need it, it would be simple to add.

Reply:
Perfect, that'll simplify things a lot.

Reply:
After thinking about it a bit, I've removed the concept of the group allocator. if this logic is not going to change in the foreseable future this new concept only adds unnecessary complexity, so I've removed it and moved the logic to the `selectGroupFromAnnocations` method and added the test directly on the `api/installations/create` tests. There's still the thing with the SQLs. Not sure if you wanted me to approach that on this PR, another one or just disregard that entirely.